chore(sdk): Fix a warning message

ysk24ok opened the pull request:
**Description of your changes:**

This PR fixes the problem that importing `kfp.v2.components.create_component_from_func` fails although `kfp.components.create_component_from_func_v2()` shows a warning saying "Please use `kfp.v2.components.create_component_from_func()` instead.".
https://github.com/kubeflow/pipelines/blob/1.7.1/sdk/python/kfp/components/_python_op.py#L875-L880

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 
<!--
   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.
-->

ji-yaqi commented:
Thanks, can you sign the cla? @ysk24ok

ysk24ok commented:
@ji-yaqi Thank you. I've signed the CLA. Could you re-run the check?

ji-yaqi commented:
@ysk24ok  Please refer to: https://github.com/kubeflow/pipelines/pull/6911/checks?check_run_id=4223191791

ji-yaqi commented:
/ok-to-test

chensun commented on sdk/python/kfp/v2/components/__init__.py:
```diff
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from kfp.v2.components.component_factory import create_component_from_func
```
Sorry, we actually don't want to include this in v2. This will not create a "v2 experience". We expect users to always use `@component` instead.

chensun commented:
> I've signed the CLA with the email address associated with [b5d3f13](https://github.com/kubeflow/pipelines/commit/b5d3f13720712c5fa223b88862753351b712c1b7) . Because https://github.com/kubeflow/pipelines/pull/6911/checks?check_run_id=4223191791 says ` Once you've signed (or fixed any issues), please request a rescan from a repository manager.`, I asked you to re-rerun the check.
> 
> Is there anything I should do? Sorry for the inconvenience.

I refreshed that, it looks good now.

chensun commented:
> **Description of your changes:**
> 
> This PR fixes the problem that importing `kfp.v2.components.create_component_from_func` fails although `kfp.components.create_component_from_func_v2()` shows a warning saying "Please use `kfp.v2.components.create_component_from_func() instead.`". https://github.com/kubeflow/pipelines/blob/1.7.1/sdk/python/kfp/components/_python_op.py#L875-L880
> 
> **Checklist:**
> 
> * [x]  The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).



> kfp.v2.components.create_component_from_func

The warning was incorrect, but in the current code from head, you won't be able to even import `kfp.v2.components.create_component_from_func`. If you like, you can help us fix the warning message in the `sdk/release-1.8` release branch, the change will go into the next SDK 1.8.* release.

ysk24ok commented:
@chensun Thank you for your review.

> The warning was incorrect, but in the current code from head, you won't be able to even import kfp.v2.components.create_component_from_func. If you like, you can help us fix the warning message in the sdk/release-1.8 release branch, the change will go into the next SDK 1.8.* release.

I've reverted b5d3f13 in 189aba9 and fixed the warning message in d830a3c.
When I changed the base branch from `master` to `sdk/release-1.8` , the diff  got messed up so I put it back in place.
After this PR is merged, I'm gonna create an another PR to cherry-pick d830a3c to `sdk/release-1.8`. Is that OK?

ysk24ok commented:
/retest

chensun commented:
> @chensun Thank you for your review.
> 
> > The warning was incorrect, but in the current code from head, you won't be able to even import kfp.v2.components.create_component_from_func. If you like, you can help us fix the warning message in the sdk/release-1.8 release branch, the change will go into the next SDK 1.8.* release.
> 
> I've reverted [b5d3f13](https://github.com/kubeflow/pipelines/commit/b5d3f13720712c5fa223b88862753351b712c1b7) in [189aba9](https://github.com/kubeflow/pipelines/commit/189aba9a1dc6128788c091686c5a5da851be0e19) and fixed the warning message in [d830a3c](https://github.com/kubeflow/pipelines/commit/d830a3c2f4773c019912e1c13742c9ee7ae5f277). When I changed the base branch from `master` to `sdk/release-1.8` , the diff got messed up so I put it back in place. After this PR is merged, I'm gonna create an another PR to cherry-pick [d830a3c](https://github.com/kubeflow/pipelines/commit/d830a3c2f4773c019912e1c13742c9ee7ae5f277) to `sdk/release-1.8`. Is that OK?

Oh, I actually forgot the warning is still in master branch, so yes, we should merge this in master first and cherry pick it in to `sdk/release-1.8`. Thank you!

chensun commented:
/retest
/lgtm
/approve

ysk24ok commented:
/retest

ysk24ok commented:
I have no idea why `kubeflow-pipelines-components-google-cloud-python38` test fails...
This PR is all about fixing the warning message.

chensun commented:
> I have no idea why `kubeflow-pipelines-components-google-cloud-python38` test fails... This PR is all about fixing the warning message.

@SinaChavoshi can you help take a look?

chensun commented:
/retest-required

chensun commented:
Actually I don't see why `kubeflow-pipelines-components-google-cloud-python38` should even be triggered with this PR. And per its run history, the test seems to be broken for a while. 
Will ignore the test.

@ysk24ok can you help with the cherrypick once this is merged? Thanks!
